core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -425,10 +425,9 @@ private[spark] class TaskSchedulerImpl(
           }
         }
       } else {
-        // No task sets are active but we still got an error. Just exit since this
-        // must mean the error is during registration.
-        // It might be good to do something smarter here in the future.
-        throw new SparkException(s"Exiting due to error from cluster scheduler: $message")
+        // No task sets are active but we still got an error. Just log the error
+        // since this must mean the error is during registration.
+        logError(s"Exiting due to error from cluster scheduler: $message")
```

Comment:
it's a fishy change if you look at how `error()` is called, https://github.com/apache/spark/blob/d5a9af3230925c347d0904fe7f2402e468e80bc8/core/src/main/scala/org/apache/spark/scheduler/cluster/SparkDeploySchedulerBackend.scala#L122

By changing this, SparkContext will not be stopped in case of error

ping the author of https://github.com/apache/spark/pull/5492 @srowen for double check